cmd/something/apicheck.go

```diff
@@ -0,0 +1,1051 @@
+	pathANX           = "https://anxv3.docs.apiary.io/"
+	pathPoloniex      = "https://docs.poloniex.com/#changelog"
+	pathIbBit         = "https://api.itbit.com/docs"
+	pathBTCMarkets    = "https://api.btcmarkets.net/openapi/info/index.yaml"
+	pathEXMO          = "https://exmo.com/en/api/"
+	pathBitstamp      = "https://www.bitstamp.net/api/"
+	pathHitBTC        = "https://api.hitbtc.com/"
+	pathBitflyer      = "https://lightning.bitflyer.com/docs?lang=en"
+	pathLakeBTC       = "https://www.lakebtc.com/s/api_v2"
+	pathKraken        = "https://www.kraken.com/features/api"
+	pathGetAllLists   = "https://api.trello.com/1/boards/%s/lists?cards=none&card_fields=all&filter=open&fields=all&key=%s&token=%s"
+	pathNewCard       = "https://api.trello.com/1/cards?%s&key=%s&token=%s"
+	pathChecklists    = "https://api.trello.com/1/checklists/%s/checkItems?%s&key=%s&token=%s"
+	apiKey            = ""
+	apiToken          = ""
+	updateCardID      = "5dfc54b96da13a6ac5ceca97"
+	updateChecklistID = "5dfc5a5377835d0ba025787a"
+)
+
+var verbose bool
+
+func main() {
+	flag.BoolVar(&verbose, "verbose", false, "increases logging verbosity for GoCryptoTrader")
+	flag.Parse()
+	updates, err := CheckUpdates(file)
+	if err != nil {
+		log.Println(err)
+	}
+	log.Println(updates)
+	// update - exchange name / apitype / reposiroty
+}
+
+// GetSha gets the sha of the latest commit
+func GetSha(repoPath string) (ShaResponse, error) {
+	var resp ShaResponse
+	finalPath := fmt.Sprintf(path, repoPath)
+	if verbose {
+		log.Println(fmt.Sprintf(path, repoPath))
+	}
+	return resp, common.SendHTTPGetRequest(finalPath, true, false, &resp)
+}
+
+// CheckExistingExchanges checks if the given exchange exists
+func CheckExistingExchanges(fileName, exchName string) ([]ExchangeInfo, bool, error) {
+	var resp bool
+	var data []ExchangeInfo
+	var err error
+	data, err = ReadFileData(fileName)
+	if err != nil {
+		return data, resp, err
+	}
+	for x := range data {
+		if data[x].Name == exchName {
+			resp = true
+			break
+		}
+	}
+	return data, resp, nil
+}
+
+// ReadFileData reads the file data for the json file
+func ReadFileData(fileName string) ([]ExchangeInfo, error) {
+	var data []ExchangeInfo
+	jsonFile, err := os.Open(fileName)
+	if err != nil {
+		return data, err
+	}
+	defer jsonFile.Close()
+	byteValue, err := ioutil.ReadAll(jsonFile)
+	if err != nil {
+		return data, err
+	}
+	json.Unmarshal(byteValue, &data)
+	return data, nil
+}
+
+// CheckUpdates checks Updates.json for all the existing exchanges
+func CheckUpdates(fileName string) ([]string, error) {
+	var resp []string
+	data, err := ReadFileData(fileName)
+	if err != nil {
+		return resp, err
+	}
+	for x := range data {
+		switch data[x].CheckType {
+		case github:
+			sha, err := GetSha(data[x].Data.GitHubData.Repo)
+			if err != nil {
+				return resp, err
+			}
+			if sha.ShaResp != data[x].Data.GitHubData.Sha {
+				if verbose {
+					log.Printf("%s api needs to be updated", data[x].Name)
```

Comment:
Parse in old and new checksum with last updated and a URL to the actual API docs. I wouldn't have this verbose.